snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy

```diff
@@ -95,20 +96,20 @@ for (String typeId : allowedRanges.keySet()) {
 	final Set<String> clauses = Sets.newHashSet();
 	
 	for (String rangeClause : rangeClauses) {
-		String clause = rangeClause;
-		
-		if (clause.contains("|")) {
-			clause = rangeClause.substring(0, rangeClause.indexOf('|')).strip();
-		}
-		
 		try {
-			eclParser.parse(clause);
-		} catch (Exception e) {
-			ctx.log().warn("Could not parse clause ${clause}");
+			eclParser.parse(rangeClause);
+		} catch (SyntaxException e) {
+			ctx.log().warn("Could not parse clause ${rangeClause}");
 			ctx.log().warn(e.getMessage());
 			continue;
 		}
 		
+		String clause = rangeClause;
+		
+		if (clause.contains("|")) {
+			clause = rangeClause.substring(0, rangeClause.indexOf('|')).strip();
```

Comment:
The parsed representation above could be put to good use to remove all human-readable labels from the expression. I'm afraid the string cutting logic will break the moment someone adds a range clause with a refinement or filter, eg. `<< 404684003 |Clinical finding| {{ C definitionStatus = primitive }}`.